Library/Formula/cryptol.rb

```diff
@@ -0,0 +1,42 @@
+require "language/haskell"
+
+class Cryptol < Formula
+  include Language::Haskell::Cabal
+
+  homepage "http://www.cryptol.net/"
+  url "https://github.com/GaloisInc/cryptol/archive/v2.2.0.tar.gz"
+  sha256 "90d2cbe35db8b2a9fcd78eaa2c08ab0cd81641a30949ab855dde71d17429d3ee"
+  head "https://github.com/GaloisInc/cryptol.git"
+
+  depends_on "ghc" => :build
+  depends_on "cabal-install" => :build
+  depends_on "cvc4"
+
+  def install
+    cabal_sandbox do
+      if build.stable?
+        # When we're building from the stable tarball, we have to
+        # provide the hash ourselves. Update this when updating the
+        # stable release.
+        system "make", "REV=0536d0f", "PREFIX=#{prefix}", "install"
```

Comment:
How long has it been since you ran `brew update`? Since https://github.com/Homebrew/homebrew/commit/359e972d4a75182ace2557e94d1a586ae66e7ad7 we stage the entire git repository (.git and all) into the build directory... no hacks should be needed.